tests/templates/test_subroutines/test_select_pauli_rot.py

```diff
@@ -130,15 +130,37 @@ def test_correctness(self, angles, rot_axis):
 
         assert np.isclose(1.0, output[0])
 
-    def test_decomposition(self):
+    @pytest.mark.parametrize("n", [1, 2, 3, 4])
+    @pytest.mark.parametrize("axis", "XYZ")
+    @pytest.mark.parametrize("batch_dim", [None, 1, 3])
+    def test_decomposition(self, n, axis, batch_dim):
         """Test that the correct gates are added in the decomposition"""
 
+        shape = (2**n,) if batch_dim is None else (batch_dim, 2**n)
+        x = np.random.random(shape)
```

Comment:
Could you set a local seed? I think we avoid having non-deterministic tests

Reply:
Done via the `seed` fixture :) :heavy_check_mark: